Check for req.body parsed as a string

Aweary opened the pull request:
`req.body` may be a string if parsed with something like `body-parser`. If it is, this creates the proper object shape used by `getGraphQLParams`

Fixes https://github.com/graphql/express-graphql/issues/10

leebyron commented:
Awesome, thank you for digging into this! Would you mind also adding a test similar to and next to https://github.com/graphql/express-graphql/blob/master/src/__tests__/http-test.js#L461 which illustrates the repro for #10? That will ensure this never regresses!

leebyron commented:
Another question, in the case that a body has already been parsed have request headers been checked as well? The existing parsed-object query expects request properties in a form that express-graphql expects, but when parsed as a string how can we ensure the request was intended to be a posted query? Perhaps in this condition we should check the content type header to see if `application/graphql` was sent, and only fulfill as a query in that case?

Aweary commented:
The request header wouldn't have been validated at that point. If we only want to respond to pared requests with a content-type of `application/graphql` then we do something like

```js
if (typeof req.body === 'string') {
      return req.headers['content-type'] === 'application/graphql'
            ? next(null, { query: req.body })
            : next(new Error('...'));
  }
```

Assuming you want to pass an error on to `next` if the header isn't correct.

leebyron commented:
I think we should mirror the current behavior of checking for the header and otherwise parsing nothing (no error). Ideally we can recycle `typeInfo` by moving the check you're doing to below that line.

Aweary commented:
Got it, I moved the check below `typeInfo` and used that to determine the return value. I also added that unit test. I tried to keep it as close to the other tests as possible, but let me know if its not up to snuff.

Aweary commented:
@leebyron any word on this?

leebyron commented:
I just squashed your refs and merged in e6fb666f87bddd383dde571f89871b2dd66f3006

Thanks for spotting this!
